worlds/stardew_valley/__init__.py

```diff
@@ -441,15 +441,43 @@ def fill_slot_data(self) -> Dict[str, Any]:
 
     def collect(self, state: CollectionState, item: StardewItem) -> bool:
         change = super().collect(state, item)
-        if change:
-            state.prog_items[self.player][Event.received_walnuts] += self.get_walnut_amount(item.name)
-        return change
+        if not change:
+            return False
+
+        player_state = state.prog_items[self.player]
+
+        received_progression_count = player_state[Event.received_progression_item]
+        received_progression_count += 1
```

Comment:
Will this count all events as well? Does it have a chance of causing trouble down the line if we add a lot of events?

Reply:
It does count all events placed into the world yes. Does it have a change of causing trouble? I don't see how it could, but maybe you have something in mind?

Reply:
I always have that concept in mind of maybe replacing all in-game item rules (ex: wood, stone, iron ore) with events instead. The first time I tried it, it didn't improve performance, but I still feel like maybe a half-and-half approach might lead to something eventually

Reply:
I previously did some testing with ores and bars, results were mitigated... Anyway, I don't think changing rules to events would create any fundamental problem with the progression item count, since events would also be added to the total. Maybe just balance issues that will need to be investigated once that happen.